src/OpenSidescan/utilities/FileLockUtils.h

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2020 © Centre Interdisciplinaire de développement en Cartographie des Océans (CIDCO), Tous droits réservés
+ */
+
+/* 
+ * File:   FileLockUtils.h
+ * Author: Christian Bouchard
+ */
+
+#ifndef FILELOCKUTILS_H
+#define FILELOCKUTILS_H
+
+#if defined(_WIN32)
+
+#include <windows.h>
+#include <stdio.h>
+struct flock flockStruct;
+
+#else
+#if defined(__linux) || defined(__linux__) || defined(linux)
+
+#include <sys/types.h>
+#include <fcntl.h>
+#include <sys/file.h>   // For flock structure
+struct flock flockStruct;
+
+#endif
+#endif
+
+class FileLockUtils {
+public:
+    
+    
+
+
+#if defined(_WIN32)
+
+static bool fileCanBeChecked(std::string filename) {
```

Comment:
- Use a pass-by-reference parameter instead of a full string copy
- Use a better name, such as fileUnlocked(...) for example.